assets/js/googlesitekit/api/index.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Get Google Site Kit data.
+ *
+ * Makes a request to this site's WordPress REST API, which will in
+ * turn make GET requests to the relevant Google services' APIs.
+ *
+ * This method automatically handles authentication, so no credentials
+ * are required to use this method.
+ *
+ * @param {string} type        The data to access. One of 'core' or 'modules'.
+ * @param {string} identifier  The data identifier, eg. a module slug like `'search-console'`.
+ * @param {string} datapoint   The endpoint to request data from.
+ * @param {Object} queryParams Query params to send with the request.
+ *
+ * @return {Promise} A promise for the `fetch` request.
+ */
+// eslint-disable-next-line no-unused-vars
+export const get = async (
+	type,
+	identifier,
+	datapoint,
+	queryParams,
+	// eslint-disable-next-line no-unused-vars
+	{ disableCache = false } = {}
+) => {
+	throw new Error( 'Not yet implemented.' );
+};
+
+/**
+ * Set Google Site Kit data.
+ *
+ * Makes a request to this site's WordPress REST API, which will in
+ * turn make requests to the relevant Google services' APIs to save
+ * the data sent in the request.
+ *
+ * This method automatically handles authentication, so no credentials
+ * are required to use this method.
+ *
+ * @param {string} type       The data to access. One of 'core' or 'modules'.
+ * @param {string} identifier The data identifier, eg. a module slug like `'adsense'`.
+ * @param {string} datapoint  The endpoint to send data to.
+ * @param {Object} data       Request data (eg. post data) to send with the request.
+ *
+ * @return {Promise} A promise for the `fetch` request.
+ */
+// eslint-disable-next-line no-unused-vars
+export const set = async (
+	type,
+	identifier,
+	datapoint,
+	data,
+	// eslint-disable-next-line no-unused-vars
+	{ disableCache = false, queryParams = {} } = {}
+) => {
+	throw new Error( 'Not yet implemented.' );
+};
+
+/**
+ * Enable/disable caching.
+ *
+ * Set the caching to on/off for the entire API library.
+ *
+ * Individual requests can still be overridden to _disable_ caching,
+ * but if caching is turned off it cannot be turned on for a specific request.
+ *
+ * @param {boolean} shouldUseCache Set to `true` to use this cache across requests; set to `false` to disable caching.
+ *
+ * @return {boolean} The current caching state (`true` for on, `false` for off).
+ */
+// eslint-disable-next-line no-unused-vars
+export const useCache = ( shouldUseCache ) => {
+	API._useCache = !! shouldUseCache;
```

Comment:
This function should also allow to simply get the current value of whether cache usage is enabled. In other words, the value should only be set if a value is actually passed. If `shouldUseCache` is `undefined`, the function should only return the current value.

Reply:
Hmm, I'm not a fan of that type of API design—it can lead to subtle bugs if an undefined value is accidentally passed and makes it more difficult to document the function. I'd rather have an `isCacheEnabled()` function or getter property; I think it's better API design than changing the behaviour of this function like that.

How about I add an `isCachingEnabled()` function?

Reply:
Sounds good. Let's name these more coherently though: `usingCache()` (for the getter) and `setUsingCache()` (for the setter). `useCache()` sounds too declarative, when reading the name you would almost think you don't need to provide any parameter, it would just "use the cache" immediately.